### PR TITLE
Extract count from each `Accum`

### PR DESCRIPTION
**Pull request by frankmcsherry**
Each `reduce::Accum` would track its own `total: Diff` counting the number of contributing records. This total is the same across all accumulations, and shouldn't need to be maintained independently if there are multiple accumulations.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  -

### CODE REVIEW

File: src/compute/src/render/reduce.rs [hunk 1]
```diff
@@ -1058,54 +1020,54 @@ where
...
                 non_nulls: if datum.is_null() { 0 } else { 1 },
             },
             AggregateFunc::Any | AggregateFunc::All => match datum {
-                Datum::True => AccumInner::Bool {
+                Datum::True => Accum::Bool {
                     trues: 1,
                     falses: 0,
                 },
-                Datum::Null => AccumInner::Bool {
+                Datum::Null => Accum::Bool {
```
Comment (tokenrove) [inline src/compute/src/render/reduce.rs:1066]: The removed comment above states

> It is important that `Accum.is_zero()` returns `false` if at least one null record has been observed. 

but wouldn't this no longer happen in this case?

Comment (frankmcsherry) [reply src/compute/src/render/reduce.rs:1066]: Yes it is no longer important, afaict. Elsewhere, `total` will be greater than zero when a non-null record has been observed, and that is what feeds into the production of `Null` for the transforms that previously relied on this.

Comment (tokenrove) [reply src/compute/src/render/reduce.rs:1066]: Ok, LGTM then if that's the case

**Review by tokenrove — APPROVED** [review]

**Comment by frankmcsherry** [comment]
Thank you for the review!

**Review by vmarcos — COMMENTED** [review]
LGTM! By the way, I tried to observe a timing difference with a variant of a CH-benCHmark query 1 adapted to the TPC-H schema at scale factor 1:
```sql
select  l_linenumber,
         sum(l_quantity) as sum_qty,
         sum(l_extendedprice) as sum_price,
         avg(l_quantity) as avg_qty,
         avg(l_extendedprice) as avg_price,
         count(*) as count_order
from     lineitem
group by l_linenumber 
order by l_linenumber;
```
However, I could not measure an effect yet. It could be that other overheads overshadow the gain at this point.

**Comment by frankmcsherry** [comment]
> However, I could not measure an effect yet. It could be that other overheads overshadow the gain at this point.

I can believe this. The `Accum` enum is something like 80 bytes before this PR, and 72 bytes after, so .. idk it might be in the noise without a very careful memory look. It's meant to be first steps on cleaning up the pile of stuff here though. :D

**Pull request merged**